client/http.go

```diff
@@ -0,0 +1,156 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/drand/drand/beacon"
+	"github.com/drand/drand/key"
+	drand "github.com/drand/drand/protobuf/drand"
+)
+
+// HTTPClient is an interface for the exercised methods of an `http.Client`,
+// or equivlanet alternative.
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
```

Comment:
Are we using this method?

Reply:
no, but i believe we expect to in the near future - e.g. specifying cache headers.